controllers/httproute_controller.go

```diff
@@ -114,9 +114,9 @@ func (r *HTTPRouteReconciler) reconcile(ctx context.Context, req ctrl.Request) e
 	// TODO(user): your logic here
 	httpLog.Info("HTTPRouteReconciler")
 
-	httpRoute := &gateway_api.HTTPRoute{}
+	httpRoute := core.HTTPRoute{}
 
-	if err := r.Client.Get(ctx, req.NamespacedName, httpRoute); err != nil {
+	if err := r.Client.Get(ctx, req.NamespacedName, &httpRoute.HTTPRoute); err != nil {
```

Comment:
looks like `&httpRoute.HTTPRoute` and `httproute.GetRuntimeObject()` do the same thing. Could we just stick to one of them? Maybe the latter is better because it is kind of hard to catch what the former means at a glance.

Reply:
Good callout, this is fixed next revision!